client/my-sites/plugins-wpcom/jetpack-plugins-panel.jsx

```diff
@@ -0,0 +1,220 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { localize } from 'i18n-calypso';
+import {
+	identity,
+	find
+} from 'lodash';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import CompactCard from 'components/card/compact';
+import PluginIcon from 'my-sites/plugins/plugin-icon/plugin-icon';
+import Button from 'components/button';
+import Gridicon from 'components/gridicon';
+import JetpackPluginItem from './jetpack-plugin-item';
+import SectionHeader from 'components/section-header';
+import SectionNav from 'components/section-nav';
+import NavTabs from 'components/section-nav/tabs';
+import NavItem from 'components/section-nav/item';
+import Search from 'components/search';
+import jetpackPlugins from './jetpack-plugins';
+
+class JetpackPluginsPanel extends Component {
+
+	static propTypes = {
+		siteSlug: PropTypes.string,
+		hasBusiness: PropTypes.bool,
+		hasPremium: PropTypes.bool,
+	};
+
+	static defaultProps = {
+		translate: identity,
+	};
+
+	constructor( props ) {
+		super( props );
+
+		this.state = {
+			selectedGroup: 'all',
+			searchTerm: '',
+		};
+	}
+
+	onNavClick( group ) {
+		this.setState( {
+			selectedGroup: group,
+		} );
+	}
+
+	onNavSearch = searchTerm => {
+		this.setState( {
+			searchTerm,
+		} );
+	}
+
+	getSearchPlaceholder() {
+		const { translate } = this.props;
+		switch ( this.state.selectedGroup ) {
+			case 'all':
+				return translate( 'Search All…', { textOnly: true } );
+			case 'engagement':
+				return translate( 'Search Engagement…', { textOnly: true } );
+			case 'security':
+				return translate( 'Search Security…', { textOnly: true } );
+			case 'appearance':
+				return translate( 'Search Appearance…', { textOnly: true } );
+			case 'writing':
+				return translate( 'Search Writing…', { textOnly: true } );
+		}
+	}
```

Comment:
this is the kind of place I like to use some abstraction to make the mapping between name and translated string more obvious. wouldn't have to do it but I like putting this outside of the component scope and into the module area

```js
const strings = ( translate = identity, name ) => get( {
	all: () => translate( 'Search All…', { textOnly: true } ),
	engagement: () => translate( 'Search Engagement…', { textOnly: true } ),
	…
}, name )();

…

const searchPlaceholder = strings( this.props.translate, this.state.selectedGroup );
```